src/Instances/TH/Lift.hs

```diff
@@ -86,8 +86,8 @@ import qualified Data.Set as Set
 import qualified Data.Tree as Tree
 
 -- Text
-import qualified Data.Text as Text
-import qualified Data.Text.Lazy as Text.Lazy
+import qualified Data.Text as Text ()
```

Comment:
Why not

```
#if !MIN_VERSION_text(1,2,4)
import qualified Data.Text as Text
import qualified Data.Text.Lazy as Text.Lazy
#endif
```

Your current PR makes no sense... you can't re-export instances.

Reply:
Apologies... that was phrased way too harsh. Here's what I meant to say is: I think it's preferable to conditionally (using CPP) add the orphan instances for `Lift Text` in this module. This way, the `th-lift-instances` can also be build with older versions of `text`, add the orphan `Lift` instances for `text < 1.2.4`, and still compile with `text >= 1.2.4`

Reply:
This is the patch I would've suggested instead: https://github.com/christiaanb/th-lift-instances/commit/97ea9cf709f9a72d4df80c8ac56e72df1c695547

Reply:
> This way, the th-lift-instances can also be build with older versions of text

btw, note that there's no need for the *same* release of `th-lift-instances` to build with old and new versions of text; The conditionality doesn't need to be handled at the CPP level; with this PR the cabal constraint solver will pick the appropriate release of `th-lift-instances` that's compatible with the respective `text` release.